pallet/ecdsa-authority/src/primitives.rs

```diff
@@ -0,0 +1,140 @@
+// This file is part of Darwinia.
+//
+// Copyright (C) 2018-2023 Darwinia Network
+// SPDX-License-Identifier: GPL-3.0
+//
+// Darwinia is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Darwinia is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Darwinia. If not, see <https://www.gnu.org/licenses/>.
+
+pub use sp_core::{ecdsa::Signature, H160 as Address, H256 as Hash};
+
+// crates.io
+use codec::{Decode, Encode};
+use scale_info::TypeInfo;
+// substrate
+use sp_io::{crypto, hashing};
+use sp_runtime::RuntimeDebug;
+
+// address(0x1)
+pub const AUTHORITY_SENTINEL: [u8; 20] =
+	[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1];
+// keccak256("ChangeRelayer(bytes4 sig,bytes params,uint256 nonce)");
+// 0x30a82982a8d5050d1c83bbea574aea301a4d317840a8c4734a308ffaa6a63bc8
+pub(crate) const RELAY_TYPE_HASH: [u8; 32] = [
+	48, 168, 41, 130, 168, 213, 5, 13, 28, 131, 187, 234, 87, 74, 234, 48, 26, 77, 49, 120, 64,
+	168, 196, 115, 74, 48, 143, 250, 166, 166, 59, 200,
+];
+// keccak256("Commitment(uint32 block_number,bytes32 message_root,uint256 nonce)");
+// 0xaca824a0c4edb3b2c17f33fea9cb21b33c7ee16c8e634c36b3bf851c9de7a223
+pub(crate) const COMMIT_TYPE_HASH: [u8; 32] = [
+	172, 168, 36, 160, 196, 237, 179, 178, 193, 127, 51, 254, 169, 203, 33, 179, 60, 126, 225, 108,
+	142, 99, 76, 54, 179, 191, 133, 28, 157, 231, 162, 35,
+];
+
+pub(crate) enum Sign {}
+impl Sign {
+	fn hash(data: &[u8]) -> [u8; 32] {
+		hashing::keccak_256(data)
+	}
+
+	pub(crate) fn eth_signable_message(chain_id: u64, spec_name: &[u8], data: &[u8]) -> Hash {
+		// \x19\x01 + keccack256(ChainIDSpecName::ecdsa-authority) + struct_hash
+		Hash(Self::hash(
+			&[
+				b"\x19\x01".as_slice(),
+				&Self::hash(&[&chain_id.to_le_bytes(), spec_name, b"::ecdsa-authority"].concat()),
```

Comment:
I change the spec from `b"46"` to `46.to_le_bytes()`. @hujw77

Reply:
Confirmed.